### PR TITLE
Box support for Indexer endpoints

### PR DESCRIPTION
**Pull request by algochoi**
This PR adds support for boxes endpoints for indexer and adds cucumber steps for testing the indexer path.

Also adds some additional comments for the algod boxes endpoints.

### CODE REVIEW

File: tests/cucumber/docker/run_docker.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ rm -rf test-harness
 rm -rf tests/cucumber/features
 
 # clone test harness
-git clone --single-branch --branch feature/box-storage https://github.com/algorand/algorand-sdk-testing.git test-harness
```
Comment (algochoi) [inline tests/cucumber/docker/run_docker.sh:10]: TODO: Revert this branch

**Review by ahangsu — APPROVED** [review]

File: src/client/v2/indexer/searchForApplicationBoxes.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   *        .SearchForApplicationBoxes(1234)
+   *        .limit(maxResults)
+   *        .nextToken(boxesPage1["next-token"])
+   *        .do();
+   * ```
+   * @param nextToken - provided by the previous results.
+   * @category query
+   */
+  nextToken(next: string) {
+    this.query.next = next;
```
Comment (tzaffi) [inline src/client/v2/indexer/searchForApplicationBoxes.ts:52]:
```suggestion
    this.query.next = "b64:" + next;
```
I'm guessing that this addresses the same issue which [I brought up in the Python SDK](https://github.com/algorand/py-algorand-sdk/pull/366/files#r948447135).

Comment (tzaffi) [reply src/client/v2/indexer/searchForApplicationBoxes.ts:52]: I verified that this works even in the empty `next` case: https://github.com/algorand/indexer/blob/tzaffi/box-search/api/test_resources/boxes.json#L886-L887

Comment (ahangsu) [reply src/client/v2/indexer/searchForApplicationBoxes.ts:52]: From the decision we made in https://github.com/algorand/algorand-sdk-testing/pull/219, we keep the `nextToken` method unchanged.

File: tests/cucumber/docker/run_docker.sh (hunk 1 above)
Comment (tzaffi) [reply tests/cucumber/docker/run_docker.sh:10]: similar issue as `go-algorand-sdk` ... need to merge in latest from `develop` branch before merging this child PR.

**Review by tzaffi — CHANGES_REQUESTED** [review]
Also seems to be needing an update from develop.

File: .test-env [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="feature/box-storage"
+SDK_TESTING_BRANCH="indexer-boxes"
```
Comment (ahangsu) [inline .test-env:3]: TODO: revert back once algorand-sdk-testing PR merge
```suggestion
SDK_TESTING_BRANCH="feature/box-storage"
```

**Review by tzaffi — APPROVED** [review]

File: src/client/v2/algod/getApplicationBoxByName.ts [hunk 4]
```diff
@@ -3,17 +3,48 @@ import HTTPClient from '../../client';
...
 import { BoxReference } from '../../../types';
 
+/**
+ * Given an application ID and the box name (key), return the value stored in the box.
+ *
+ * #### Example
+ * ```typescript
+ * const index = 1234;
+ * const boxName = Buffer.from("foo");
+ * const app = await algodClient.getApplicationBoxByName(index).name(boxName).do();
```
Comment (ahangsu) [inline src/client/v2/algod/getApplicationBoxByName.ts:13]:
```suggestion
 * const boxValue = await algodClient.getApplicationBoxByName(index).name(boxName).do();
```

**Pull request merged**